tests/lowering/matmul/test_linear.py

```diff
@@ -0,0 +1,38 @@
+import torch
+import torch_ttnn
+import pytest
+import ttnn
+
+from tests.utils import assert_with_pcc
+
+
+class LinearModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, input, weight):
+        return torch.nn.functional.linear(input, weight)
+
+
+@pytest.mark.parametrize(
+    "input_shapes",
+    [[(4, 8), (6, 8)]],
+)
+def test_linear(device, input_shapes):
+    m = LinearModule()
+    inputs = [torch.rand(shape, dtype=torch.bfloat16) for shape in input_shapes]
+    result_before = m.forward(*inputs)
+    option = torch_ttnn.TorchTtnnOption(device=device)
+    option.gen_graphviz = True
+    # The compilation is lazy, so we need to run forward once to trigger the compilation
+    m = torch.compile(m, backend=torch_ttnn.backend, options=option)
+    result_after = m.forward(*inputs)
+    option._out_fx_graphs[0].print_tabular()
+
+    # Check the graph has be rewritten and contain ttnn ops
+    nodes = list(option._out_fx_graphs[0].nodes)
+    target = [node.target for node in nodes]
+    assert target.count(ttnn.linear) == 1
```

Comment:
This test fails because it is further converted to `ttnn.permute` and `ttnn.matmul`. I'm not sure if this is the expected behavior, but it looks legit.

```
----------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------
opcode         name               target                                                                                                                                                                                                                                                                                                                                                                                                                                                                             args                               kwargs
-------------  -----------------  ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------  ---------------------------------  -------------------------------------------------------------
placeholder    arg0_1             arg0_1                                                                                                                                                                                                                                                                                                                                                                                                                                                                             ()                                 {}
placeholder    arg1_1             arg1_1                                                                                                                                                                                                                                                                                                                                                                                                                                                                             ()                                 {}
call_function  ttnn_from_torch    FastOperation(python_fully_qualified_name='ttnn.from_torch', function=<function from_torch at 0x7fd109f00700>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7fd147860700>, golden_function=<function _golden_function at 0x7fd109f003a0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7fd147860940>, is_cpp_operation=False, is_experimental=False)                               (arg1_1,)                          {'layout': ttnn_TILE_LAYOUT, 'device': ttnn_Specified_Device}
call_function  ttnn_permute       FastOperation(python_fully_qualified_name='ttnn.permute', function=<ttnn._ttnn.operations.data_movement.permute_t object at 0x7fd147c96030>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7fd147860700>, golden_function=<function _golden_function at 0x7fd109f04f70>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7fd147860940>, is_cpp_operation=True, is_experimental=False)  (ttnn_from_torch, (1, 0))          {}
call_function  ttnn_from_torch_1  FastOperation(python_fully_qualified_name='ttnn.from_torch', function=<function from_torch at 0x7fd109f00700>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7fd147860700>, golden_function=<function _golden_function at 0x7fd109f003a0>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7fd147860940>, is_cpp_operation=False, is_experimental=False)                               (arg0_1,)                          {'layout': ttnn_TILE_LAYOUT, 'device': ttnn_Specified_Device}
call_function  ttnn_matmul        FastOperation(python_fully_qualified_name='ttnn.matmul', function=<function matmul at 0x7fd109ed2160>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7fd147860700>, golden_function=<function _golden_function at 0x7fd109ecff70>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7fd147860940>, is_cpp_operation=False, is_experimental=False)                                       (ttnn_from_torch_1, ttnn_permute)  {}
call_function  ttnn_to_torch      FastOperation(python_fully_qualified_name='ttnn.to_torch', function=<function to_torch at 0x7fd109f00940>, preprocess_golden_function_inputs=<function default_preprocess_golden_function_inputs at 0x7fd147860700>, golden_function=<function _golden_function at 0x7fd109f00670>, postprocess_golden_function_outputs=<function default_postprocess_golden_function_outputs at 0x7fd147860940>, is_cpp_operation=False, is_experimental=False)                                   (ttnn_matmul,)                     {}
output         output             output                                                                                                                                                                                                                                                                                                                                                                                                                                                                             ((ttnn_to_torch,),)                {}
--------------------------------------------------------------------------------- Captured stdout teardown ----------------------------------------------------------------------------------
```